cmd/client/main.go

```diff
@@ -92,7 +94,18 @@ func loadConfig() (*config, *client.Config) {
 	}
 
 	clientCfg := client.GetDefaultConfig()
+
+	hostname, err := os.Hostname()
+	if err == nil {
+		clientCfg.Hostname = hostname
+	}
+	clientCfg.Hostname = env(envClientHost, clientCfg.Hostname)
 	clientCfg.DistroHost = env(envDistroHost, clientCfg.DistroHost)
+	clientCfg.ConsulHost = env(envConsulHost, clientCfg.ConsulHost)
+	port, err := strconv.Atoi(env(envConsulPort, string(clientCfg.ConsulPort)))
+	if err == nil {
```

Comment:
What happens if there is an error? Should it be logged / aborted immediately, or you're just deferring the decision? Besides, it is somewhat awkward to have a function signature that hides all possible errors.